packages/application-extension/src/index.tsx

```diff
@@ -428,6 +428,22 @@ const liteLogo: JupyterFrontEndPlugin<void> = {
   },
 };
 
+/**
+ * A plugin to trigger a refresh of the commands when the shell layout changes.
+ */
+const notifyCommands: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlite/application-extension:notify-commands',
+  autoStart: true,
+  optional: [ILabShell],
+  activate: (app: JupyterFrontEnd, labShell: ILabShell | null) => {
+    if (labShell) {
+      labShell.layoutModified.connect(() => {
+        app.commands.notifyCommandChanged();
```

Comment:
Not really happy about this, as this is hinting for some race conditions upstream in lab as to how the toolbar items are now populated (based on the settings).

This might also be because the lite settings endpoint sends the response very quickly (faster server roundtrip), which removes a delay JupyterLab might expect to find.

Reply:
cc @fcollonval who might know more

Reply:
I saw that line a while ago and don't like it neither. I don't have any clue why it was made. But it definitely hide explicit call at more appropriate places.
For the bug you are seeing, it points towards the toolbar being build before the notebook being set as the current widget.

Reply:
It may be more appropriate to trigger the notification when the current widget changes rather than the broader layout change signal.

Reply:
Looks like this code is 4 years old in the lab code base, so yeah something that's been around for a while.